File: src/main/java/net/gtaun/util/event/Attentions.java
package net.gtaun.util.event;

import java.util.ArrayList;
import java.util.Collection;

public class Attentions
{
	public static Attentions create()
	{
		return new Attentions();
	}
	
	public static Attentions all()
	{
		return create().clazz(Object.class);
	}
	
	
	private Collection<Object> objects;
	
	
	private Attentions()
	{
		objects = new ArrayList<>();
	}

	Collection<Object> getObjects()
	{
		return objects;
	}
	
	public Attentions object(Object target)
	{
		objects.add(target);
		return this;
	}
	
	public Attentions clazz(Class<?> clz)
	{
		objects.add(clz);
		return this;
	}
}

File: src/main/java/net/gtaun/util/event/Attentions.kt
package net.gtaun.util.event

import kotlin.reflect.KClass

class Attentions {

    val objects: MutableList<Any> = mutableListOf()

    fun `object`(target: Any): Attentions {
        objects.add(target)
        return this
    }

    fun clazz(clz: KClass<*>): Attentions =
            clazz(clz.java)

    fun clazz(clz: Class<*>): Attentions {
        objects.add(clz)
        return this
    }

    companion object {
        @JvmStatic
        fun create(): Attentions = Attentions()

        @JvmStatic
        fun all(): Attentions = create().clazz(Any::class.java)
    }
}
